scripts/deploy/deploy.js

```diff
@@ -310,9 +313,12 @@ async function main() {
   await cover.changeMasterAddress(master.address);
   await cover.changeDependentContractAddress();
 
-  await cover.addProductTypes(
-    productTypes,
-    productTypes.map(() => ''), // ipfs metadata for each product type
+  await cover.setProductTypes(
+    productTypes.map(productType => ({
+      productType,
+      productTypeId: MaxUint256,
+      ipfsMetadata: productType.descriptionIpfsHash,
+    })),
```

Comment:
This is the only place `productTypes` variable is used. Add `productTypeId` and rename the ipfs field name in the definition directly (line 26).